User Story 89659

TomArcherMsft opened the pull request:
[User Story 89659](https://dev.azure.com/msft-skilling/Content/_workitems/edit/89659)

TomArcherMsft committed:
Changed deprecated TF resource (azurerm_virtual_machine)

stemaMSFT commented on quickstart/101-vm-cluster-linux/main.tf:
```diff
@@ -0,0 +1,123 @@
+}
+
+resource "random_pet" "azurerm_virtual_network_name" {
+  prefix = "vn"
```
I could be wrong but don't we usually prefix with `vnet`, not `vn`?

stemaMSFT commented on quickstart/101-vm-cluster-linux/readme.md:
```diff
@@ -0,0 +1,28 @@
+# Azure Linux VM cluster
+
+This template deploys a Windows VM cluster on Azure.
```
```suggestion
This template deploys a Linux VM cluster on Azure.
```

stemaMSFT reviewed (changes requested):
Some minor changes then should be good @TomArcherMsft

TomArcherMsft committed:
Edits per Code Review

TomArcherMsft commented:
@stemaMSFT @lonegunmanb I've made the requested edits and ready for another review. Thanks!

lonegunmanb commented on quickstart/101-vm-cluster-linux/main.tf:
```diff
@@ -0,0 +1,123 @@
+  location              = azurerm_resource_group.rg.location
+  availability_set_id   = azurerm_availability_set.avset.id
+  resource_group_name   = azurerm_resource_group.rg.name
+  network_interface_ids = [element(azurerm_network_interface.test.*.id, count.index)]
```
Since both `count` of nic and vm are `2`, could we simplify the expression here?:

```hcl
network_interface_ids = [azurerm_network_interface.test[count.index].id]
```

lonegunmanb commented on quickstart/101-vm-cluster-linux/main.tf:
```diff
@@ -0,0 +1,123 @@
+  }
+}
+
+resource "azurerm_managed_disk" "test" {
```
I couldn't find the usage of this managed disk, missing the attachment?

lonegunmanb commented on quickstart/101-vm-cluster-linux/main.tf:
```diff
@@ -0,0 +1,123 @@
+
+  computer_name                   = "hostname"
+  admin_username                  = "testadmin"
+  admin_password                  = "Password1234!"
```
Could we use `random_password` to generate a strong password here?

lonegunmanb reviewed (changes requested):
Hi @TomArcherMsft thanks for opening this pr! A few comments.

TomArcherMsft committed:
Edits per Code Review

TomArcherMsft commented:
CC: @stemaMSFT 

@lonegunmanb Thanks for the great review. I believe I've addressed the issues you mention. Please verify that I made the edits correctly. Thanks!

TomArcherMsft committed:
Changed admin username

lonegunmanb commented on quickstart/101-vm-cluster-linux/main.tf:
```diff
@@ -0,0 +1,145 @@
+  resource_group_name  = azurerm_resource_group.rg.name
+  storage_account_type = "Standard_LRS"
+  create_option        = "Empty"
+  disk_size_gb         = "1023"
```
Why the size is not `1024` but `1023`?

TomArcherMsft replied:
> Thanks @TomArcherMsft, LGTM, only one question remains but the pr is good to merge.

Thanks, @lonegunmanb. I made the change.

lonegunmanb reviewed (approved):
Thanks @TomArcherMsft, LGTM, only one question remains but the pr is good to merge.

TomArcherMsft committed:
Made suggested change
